### DISCUSSION TITLE
How can I watch java junit test amoung relative java file

### DISCUSSION THREAD

**Discussion by GaryChung34** [nvim-neotest/neotest#363] [Q&A]
greeting, my problem is trying to use watch.toggle a test java file which is testing a function of one of my class. However, when I change and save this class file. no test run is triggle. wonder why?

For more info, after i read the doc of neotest.watch, i see that watching the relative file is dependent of LSP client. some config can be made in the watch consumer. can I config the watch consumer to watch my project directory of re-run my test? where alse can i find more information about the watch config? thanks.

i am using: 
1. jdlts LSP server in Mason 
2. neotest-java adopter

here is my neotest config:
![image](https://github.com/nvim-neotest/neotest/assets/87843133/d26d4f95-9935-48f2-91c4-5f408e2a69bd)

**Comment by rcarriga** [comment]
You need to add a `java` entry to the `symbol_queries` table (see [here](https://github.com/nvim-neotest/neotest/blob/dcdb40ea48f9c7b67a5576f6bb2e5f63ec15f2c0/lua/neotest/config/init.lua#L273-L343) for existing ones), which uses treesitter to find symbols that the watch consumer will request definition for using the LSP client ([here](https://github.com/nvim-neotest/neotest/blob/dcdb40ea48f9c7b67a5576f6bb2e5f63ec15f2c0/lua/neotest/consumers/watch/watcher.lua#L51-L68)). The queries need to match a capture group `@symbol` which will define the range of text that will be sent to the LSP server. If you're not familiar with building treesitter queries, I'd recommend checking out `:help treesitter-query`

These symbols can be anything but it's recommended to use things like imports which will link to other files and build a dependency tree of files which once they are written, the dependent tests will re-run
